### PR TITLE
Bignums arithmetics

### PR DESCRIPTION
**Pull request by JuliaGerasymenko**

### CODE REVIEW

File: task1.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+let base = 1000000;
+
+function addition(arr1, arr2) {
+  if (arr1.length < arr2.length) {
+    let tmp = arr1;
```
Comment (DzyubSpirit) [inline task1.js]: JavaScript allows to do this:
```
[a, b] = [b, a];
```

File: task1.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    arr1 = arr2;
+    arr2 = tmp;
+  }
+  if (arr2[0] === 0 && arr2.length === 1) return arr1;
+  let len1 = arr1.length;
+  let len2 = arr2.length;
+  while (arr2.length !== len1) {
+    arr2.push(0);
```
Comment (DzyubSpirit) [inline task1.js]: It's better to avoid unnecessary modifications of the input array.

File: task1.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  }
+
+  for (let i = 0; i < len1; i++) {
+    arr1[i] += arr2[i];
+    if (arr1[i] > base - 1) {
```
Comment (DzyubSpirit) [inline task1.js]:
```suggestion
    if (arr1[i] >= base) {
```

File: task1.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      arr1[i] -= base;
+      i === len1 - 1 ? arr1.push(1) : (arr1[i + 1] += 1);
+    }
+  }
+  return arr1;
+}
+
+function multiplication(arr1, arr2) {
+  if (!arr1[0] && arr1.length === 1) return arr1;
```
Comment (DzyubSpirit) [inline task1.js]:
```suggestion
  if (!arr2[0] && arr2.length === 1) return arr1;
```

File: task1.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+  // unusable for a case where arr1 = arr2
+  // while (arr2.length !== len1) {
+  //   arr2.push(0);
+  // }
+
+  let res = new Array(len2 + len1 - 1).fill(0);
+  let resLen = res.length - 1;
+  for (let i = 0; i < len1; i++) {
+    for (let j = 0; j < len1; j++) {
+      res[j + i] += use strict';
```
Comment (DzyubSpirit) [inline task1.js]: Seems like unexpected file collapse is here

**Review by DzyubSpirit — CHANGES_REQUESTED** [review]

File: task1.js (hunk 2 above)
Comment (JuliaGerasymenko) [reply task1.js]: Why these modifications are unnecessary?

File: task1.js (hunk 4 above)
Comment (JuliaGerasymenko) [reply task1.js]: arr1 and arr2 are the same

Comment (DzyubSpirit) [reply task1.js]: My bad

File: task1.js (hunk 2 above)
Comment (DzyubSpirit) [reply task1.js]: Because there is a way to solve the problem without them

File: task1.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    if (len1 < len2) {
+      while (arr2.length !== len1) {
+        arr1.push(0);
+      }
+    } else {
+      while (arr2.length !== len1) {
+        arr2.push(0);
+      }
+    }
+  }
```
Comment (DzyubSpirit) [inline task1.js]: This is unnecessary.

File: task1.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+  let res = new Array(len2 + len1 - 1).fill(0);
+  let resLen = res.length - 1;
+  for (let i = 0; i < len1; i++) {
```
Comment (DzyubSpirit) [inline task1.js]:
```suggestion
  for (let i = 0; i < len2; i++) {
```

File: task1.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    for (let j = 0; j < len1; j++) {
+      res[j + i] += arr2[i] * arr1[j];
+      if (res[j + i] > base - 1) {
```
Comment (DzyubSpirit) [inline task1.js]:
```suggestion
      if (res[j + i] >= base) {
```

File: task1.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+        res[j + i + 1] === undefined
+          ? res.push(divider)
+          : (res[j + i + 1] += divider);
+      }
+    }
+  }
+
+  while (!res[res.length - 1]) {
+    res.pop();
+  }
```
Comment (DzyubSpirit) [inline task1.js]: If you remove the addition of zeros earlier then `res.pop()` never be called.

File: task1.js [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+  let a1 = [t1];
+  let a2 = [t2];
+  let count = 3;
+
+  for (let i = 3; i <= n; i++) {
+    a1 = addition(a1, multiplication(a2,a2));
+    if (i !== n) {
+      let tmp = a1;
+      a1 = a2;
+      a2 = tmp;
```
Comment (DzyubSpirit) [inline task1.js]: ```[a1, a2] = [a2, a1]```

File: task1.js [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    }
+  }
+
+  return (a1.length > 1
+    ? a1.map((el, i) => {
+        el = el.toString();
+        while (el.length < 6 && i !== a1.length - 1) {
```
Comment (DzyubSpirit) [inline task1.js]: nit:
```javascript
if (i === a1.length - 1) return el;
while (el.length < baseLog) {
```
Where ```baseLog = Math.log10(base)``` can be included right after definition of `base`.

**Review by DzyubSpirit — CHANGES_REQUESTED** [review]

File: task1.js [hunk 12]
```diff
@@ -1,48 +1,48 @@
 'use strict';
 
-let base = 1000000;
+let base = 10000;
+let baseLog = Math.log10(10*base);
```
Comment (DzyubSpirit) [inline task1.js]: Why do you use `Math.log10(10*base)` instead of `Math.log10(base)`?

File: task1.js (hunk 7 above)
Comment (DzyubSpirit) [reply task1.js]: Are you going to apply this suggestion or is it unnecessary?

Comment (JuliaGerasymenko) [reply task1.js]: Concreate for Fibonacci task `len1` and` len2` are equal. So I stashed them and declare an additional variable `len`

File: task1.js [hunk 13]
```diff
@@ -36,11 +36,11 @@ function multiplication(arr1, arr2) {
   //     while (arr2.length !== len1) arr2.push(0);
   //   }
 
-  let res = new Array(len2 + len1 - 1).fill(0);
+  let res = new Array(2*len - 1).fill(0);
```
Comment (DzyubSpirit) [inline task1.js]: Why did you do that?

File: task1.js (hunk 7 above)
Comment (DzyubSpirit) [reply task1.js]: True. But I think that the long multiplication should be more abstract and work with arrays of different sizes.

**Review by DzyubSpirit — APPROVED** [review]

**Pull request merged**